src/addons/matchrecursive.js

```diff
@@ -179,10 +180,19 @@ export default (XRegExp) => {
                         break;
                     }
                 }
-            } else {
+            } else if (unbalancedDelimiters === 'error') {
                 const delimSide = rightMatch ? 'right' : 'left';
                 const errorPos = rightMatch ? delimStart : outerStart;
                 throw new Error(`Unbalanced ${delimSide} delimiter found in string at position ${errorPos}`);
+            } else if (unbalancedDelimiters === 'text') {
+                if (rightMatch) {
+                    rightMatch = null;
+                } else {
+                    delimEnd = XRegExp.exec(str, left, outerStart)[0].length + outerStart;
+                    openTokens = 0;
```

Comment:
this is the case when it's an unbalanced left delimiter. We move `delimEnd` back to right after the unbalanced left delimiter and set `openTokens` to 0 to "restart" the loop and unbalanced delimiter then is treated as raw text